### ISSUE TITLE
Feedback and suggestions?

### ISSUE THREAD

**Issue by Hodorcoder** [ZeusNoTZues/Competetive-Programming-Q-A#2]
I don't know if you want feedback and suggestions, or just bug reports. If you don't want feedback or suggestions then just let me know and I'll stop.

I was looking at: https://github.com/ZeusNoTZues/Competetive-Programming-Q-A/blob/master/Problems-and-Solutions/Easy/CodeForces/%23263A%20Beautiful%20Matrix.cpp

Have you considered implementing this using a single-dimension array? Personally I think it makes it more readable, but maybe I'm weird

E.g.

    int main(void)
    {
        int M[5*5], idx_one;
        
        for (int i = 0; i < 25; i++) {
            std::cin >> M[i];
            if (M[i])
                idx_one = i;
        }
    
        int r = idx_one - 12;   // 12 is idx for the middle of matrix
        std::cout << (r < 0 ? -r : r) << std::endl;
            
        return 0;
    }

**Comment by ZeusNoTZues** [comment]
I'd like to hear anything that you can provide me with. So, be it feedback and suggestions or bug reports, feel free to let me know what's up.

**Comment by Hodorcoder** [comment]
Well, I was looking at https://github.com/ZeusNoTZues/Competetive-Programming-Q-A/blob/master/Problems-and-Solutions/Easy/CodeForces/%23263A%20Beautiful%20Matrix.cpp today and it occurred to me that if I implemented that solution I would have used a 1-dimensional array and not a 2-dimensional array. I'm not saying that the 1-dimensional approach is better or anything but have you considered implementing it using a 1-dimensional array? To _me_ the 1-dimensional approach results in more readable code but of course that's subjective.

**Comment by ZeusNoTZues** [comment]
How about [this](https://github.com/ZeusNoTZues/Competetive-Programming-Q-A/blob/master/Problems-and-Solutions/Easy/CodeForces/%23263A%20Beautiful%20Matrix.cpp)?

**Comment by Hodorcoder** [comment]
Yeah, that's more like how I did it. For some reason I hate 2d arrays. But whatever feels most natural is the way to go ;)

    Moves = abs((2 - i / 5) + (2 - i % 5));

**Comment by Hodorcoder** [comment]
RE: https://github.com/ZeusNoTZues/Competetive-Programming-Q-A/blob/ef2ff4fa158d65b99e3169bdb6b5b5ce47de273b/Problems-and-Solutions/CodeForces/%231104A%20Splitting%20into%20Digits.cpp

Doesn't seem to be correct (but perhaps I'm missing something)

$ echo 27 | ./a.out 
Output
`27`
`1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1 1`

$ echo 27 | ./a.out 
Expected
`3`
`9 9 9`

**Comment by ZeusNoTZues** [comment]
It need not necessarily be the same as the expected output. The problem just asked to split the number in a sensible way. Printing 1's was the easiest. Also, you could test the same code on their website with many other test cases when submitting to check if your code is correct. Everything I commit here is an accepted solution :)

**Comment by Hodorcoder** [comment]
Ok, well that's what I missed I guess :)
